components/sensor/power_ina219/power_ina219.go

```diff
@@ -0,0 +1,229 @@
+// Package ina219 implements an ina219 voltage/current/power monitor sensor -
+// typically used for battery state monitoring.
+// Datasheet can be found at: https://www.ti.com/lit/ds/symlink/ina219.pdf
+// Example repo: https://github.com/periph/devices/blob/main/ina219/ina219.go
+package ina219
+
+import (
+	"context"
+	"encoding/binary"
+	"fmt"
+
+	"github.com/edaniels/golog"
+	"go.viam.com/utils"
+
+	"go.viam.com/rdk/components/board"
+	"go.viam.com/rdk/components/generic"
+	"go.viam.com/rdk/components/sensor"
+	"go.viam.com/rdk/config"
+	"go.viam.com/rdk/registry"
+	"go.viam.com/rdk/resource"
+	rdkutils "go.viam.com/rdk/utils"
+)
+
+var modelname = resource.NewDefaultModel("power_ina219")
+
+const (
+	milliAmp             = 1000 * 1000 // milliAmp = 1000 microAmpere * 1000 nanoAmpere
+	milliOhm             = 1000 * 1000 // milliOhm = 1000 microOhm * 1000 nanoOhm
+	defaultI2Caddr       = 0x40
+	senseResistor        = 100 * milliOhm  // .1 ohm = 100 * milliOhm * microOhm * nanoOhm
+	maxCurrent           = 3200 * milliAmp // 3.2 amp = 3200 * milliAmpere * microAmpere * nanoAmpere
```

Comment:
:+1: This is very clever! Nice choice.

Do the comments still need to be so long? I can go either way.